Neos.ContentRepositoryRegistry/Classes/ContentRepositoryRegistry.php

```diff
@@ -242,29 +245,40 @@ private function buildPropertySerializer(ContentRepositoryId $contentRepositoryI
     /** @param array<string, mixed> $contentRepositorySettings */
     private function buildProjectionsFactory(ContentRepositoryId $contentRepositoryId, array $contentRepositorySettings): ProjectionsAndCatchUpHooksFactory
     {
-        (isset($contentRepositorySettings['projections']) && is_array($contentRepositorySettings['projections'])) || throw InvalidConfigurationException::fromMessage('Content repository "%s" does not have projections configured, or the value is no array.', $contentRepositoryId->value);
+        (is_array($contentRepositorySettings['contentRepositoryProjection'] ?? null)) || throw InvalidConfigurationException::fromMessage('Content repository "%s" does not have the contentRepositoryProjection configured, or the value is no array.', $contentRepositoryId->value);
+        (is_array($contentRepositorySettings['projections'] ?? [])) || throw InvalidConfigurationException::fromMessage('Content repository "%s" expects projections configured as array.', $contentRepositoryId->value);
         $projectionsFactory = new ProjectionsAndCatchUpHooksFactory();
+        $this->registerProjection($contentRepositorySettings['contentRepositoryProjection'], $projectionsFactory, ContentRepositoryProjectionFactoryInterface::class, 'contentRepositoryProjection', $contentRepositoryId);
         foreach ($contentRepositorySettings['projections'] as $projectionName => $projectionOptions) {
             if ($projectionOptions === null) {
                 continue;
             }
-            $projectionFactory = $this->objectManager->get($projectionOptions['factoryObjectName']);
-            if (!$projectionFactory instanceof ProjectionFactoryInterface) {
-                throw InvalidConfigurationException::fromMessage('Projection factory object name for projection "%s" (content repository "%s") is not an instance of %s but %s.', $projectionName, $contentRepositoryId->value, ProjectionFactoryInterface::class, get_debug_type($projectionFactory));
+            $this->registerProjection($projectionOptions, $projectionsFactory, ProjectionFactoryInterface::class, $projectionName, $contentRepositoryId);
+        }
+        return $projectionsFactory;
+    }
+
+    /**
+     * @param class-string<ProjectionFactoryInterface<ProjectionInterface<ProjectionStateInterface>>> $expectedProjectionFactory
+     */
+    private function registerProjection(mixed $projectionOptions, ProjectionsAndCatchUpHooksFactory $projectionsFactory, string $expectedProjectionFactory, string $projectionName, ContentRepositoryId $contentRepositoryId): void
+    {
+        (is_array($projectionOptions)) || throw InvalidConfigurationException::fromMessage('Projection "%s" (content repository "%s") must be configured as array got %s', $projectionName, $contentRepositoryId->value, get_debug_type($projectionOptions));
```

Comment:
`(is_array($projectionOptions)) || throw InvalidConfigu...`

YIKES

also why is it mixed in the signature then

Reply:
later alligator